src/main/kotlin/org/jetbrains/research/transformations/Main.kt

```diff
@@ -1,5 +1,48 @@
 package org.jetbrains.research.transformations
 
+import com.github.gumtreediff.gen.python.PythonTreeGenerator
+import com.github.gumtreediff.tree.ITree
+import com.github.gumtreediff.tree.TreeContext
+import java.io.File
+import org.jetbrains.research.transformations.Transformations.*
+
+fun setupParser() {
+    //copy parser file into /tmp directory, required by Gumtree
+    val sourcePath :String = "src/main/resources/pythonparser"
+    val targetPath :String = "/tmp/pythonparser"
+    val pythonparserFile :File = File(sourcePath)
+
+    try {
+        val targetFile :File = File(targetPath)
+        pythonparserFile.copyTo(targetFile)
+        val command = arrayOf("chmod", "+x", targetFile.getAbsolutePath())
+        val b = ProcessBuilder(*command)
+        b.directory(targetFile.parentFile)
+        b.start()
+    } catch (e: kotlin.io.FileAlreadyExistsException) {
+        println("Pythonparser file already exists in ${targetPath} directory")
+    }
+
+    // add pythonparser's path into system path
+    System.setProperty("gt.pp.path", targetPath)
+}
+
+fun printTreeFromRoot(tree : ITree) {
+    for (node in tree.children) {
+        println("${node.type}, ${node.label}, ${node.getMetadata(node.label)}")
+        printTreeFromRoot(node)
+    }
+}
```

Comment:
Please, don't use **for**. You can do something like this:
```
    tree.children.forEach {
        println("${it.type}, ${it.label}, ${it.getMetadata(it.label)}")
        printTreeFromRoot(it)
    }
```

Also, such printing is usually called "prettyPrint" (see https://en.wikipedia.org/wiki/Prettyprint), so it's better to rename it. And to make it even more idiomatic kotlin styled, you can **extend** interface ITree with prettyPrint:
```
fun ITree.prettyPrint() {
    children.forEach {
        println("${it.type}, ${it.label}, ${it.getMetadata(it.label)}")
        it.prettyPrint()
    }
}
```

Reply:
Probably we should explain it in the coming meeting. Because maybe it is not clear for @RasAV why he should not use it. And it is a good ides again, thanks!